src/PaymentGateways/Actions/RegisterPaymentGateways.php

```diff
@@ -140,7 +140,10 @@ private function addGatewayDataToPayPalCommerce()
                 PayPalCommerce::id()
             ),
             function ($gatewayData) {
-                $gatewayData['paypalOrder'] = (new GetPayPalOrderFromRequest())();
+                if(!isset($gatewayData['payPalOrderId'])) {
+                    $gatewayData['payPalOrderId'] = give_clean($_POST['payPalOrderId']);
+                }
+                $gatewayData['paypalOrder'] = (new GetPayPalOrderFromRequest())($gatewayData['payPalOrderId']);
```

Comment:
@kjohnson can we clean this up a bit while we're here?

`GetPayPalOrderFromRequest`(by it's name) should either have all the request logic for retrieving the `payPalOrderId` or we can do the request logic here and just call `give(PayPalOrder::class)->getOrder($paypalOrderId)` directly without the action.

Both ways should be using a try/catch as well.